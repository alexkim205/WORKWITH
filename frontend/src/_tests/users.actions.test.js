import createMockStore from "../_config/mockStore.config";
import usersActions from "../_actions/users.actions";
import testConstants from "../_constants/test.constants";

const {
  users: { user, loginCredentials, newUser, updateUser }
} = testConstants;

describe("User Actions", () => {
  let store;
  beforeAll(() => {
    store = createMockStore();
  });
  afterEach(() => {
    store.clearActions();
  });
  it("it should GET users", async () => {
    await store.dispatch(usersActions.getUsers());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("it should GET a user", async () => {
    await store.dispatch(usersActions.getUser(user._id));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("it should LOGIN a user", async () => {
    await store.dispatch(
      usersActions.login(loginCredentials.email, loginCredentials.password)
    );
    expect(store.getActions()).toMatchSnapshot();
  });
  it("it should REGISTER a user", async () => {
    await store.dispatch(usersActions.register(newUser));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("it should UPDATE a user", async () => {
    await store.dispatch(usersActions.updateUser(user._id, updateUser));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("it should DELETE a note", async () => {
    await store.dispatch(usersActions.deleteUser(user._id));
    expect(store.getActions()).toMatchSnapshot();
  });
});
