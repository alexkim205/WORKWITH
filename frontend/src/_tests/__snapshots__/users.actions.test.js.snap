// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Actions it should DELETE a note 1`] = `
Array [
  Object {
    "type": "users/delete-user-pending",
  },
  Object {
    "payload": undefined,
    "type": "users/delete-user-success",
  },
]
`;

exports[`User Actions it should GET a user 1`] = `
Array [
  Object {
    "type": "users/get-user-pending",
  },
  Object {
    "payload": Object {
      "user": Object {
        "__v": 0,
        "_id": "5e0bb5c61c90231c34596f83",
        "createdAt": "2019-12-31T20:55:34.424Z",
        "deleted": false,
        "email": "alexkim@dev.com",
        "hash": "8a68a2ca14e41925e745d3f5a14550a427aa05c4bc00248e63dc45d12e10e2e3f7e1e09822dd544c1f808dd71eb1aa6538b96408ff73d2ecfd9b54b204fad705",
        "name": "Alex Kim",
        "salt": "563ad7a2ec596b8bdca6cc5b3040e302",
        "updatedAt": "2019-12-31T20:55:34.424Z",
      },
    },
    "type": "users/get-user-success",
  },
]
`;

exports[`User Actions it should GET users 1`] = `
Array [
  Object {
    "type": "users/get-users-pending",
  },
  Object {
    "payload": Object {
      "users": Array [
        Object {
          "__v": 0,
          "_id": "5e0bb5c61c90231c34596f83",
          "createdAt": "2019-12-31T20:55:34.424Z",
          "deleted": false,
          "email": "alexkim@dev.com",
          "hash": "8a68a2ca14e41925e745d3f5a14550a427aa05c4bc00248e63dc45d12e10e2e3f7e1e09822dd544c1f808dd71eb1aa6538b96408ff73d2ecfd9b54b204fad705",
          "name": "Alex Kim",
          "salt": "563ad7a2ec596b8bdca6cc5b3040e302",
          "updatedAt": "2019-12-31T20:55:34.424Z",
        },
      ],
    },
    "type": "users/get-users-success",
  },
]
`;

exports[`User Actions it should LOGIN a user 1`] = `
Array [
  Object {
    "type": "users/login-success",
  },
  Object {
    "payload": Object {
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1Nzg1NDYxMjUsImlhdCI6MTU3Nzk0MTMyNX0.ksgtQ8uDQ3322tOhQ4jIkzZjH6B0luhLA1NRKGCG00w",
      "user": Object {
        "__v": 0,
        "_id": "5e0bb5c61c90231c34596f83",
        "createdAt": "2019-12-31T20:55:34.424Z",
        "deleted": false,
        "email": "alexkim@dev.com",
        "hash": "8a68a2ca14e41925e745d3f5a14550a427aa05c4bc00248e63dc45d12e10e2e3f7e1e09822dd544c1f808dd71eb1aa6538b96408ff73d2ecfd9b54b204fad705",
        "name": "Alex the Great",
        "salt": "563ad7a2ec596b8bdca6cc5b3040e302",
        "updatedAt": "2019-12-31T21:00:15.298Z",
      },
    },
    "type": "users/login-success",
  },
]
`;

exports[`User Actions it should REGISTER a user 1`] = `
Array [
  Object {
    "type": "users/register-pending",
  },
  Object {
    "payload": Object {
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1Nzg1NDYxMjUsImlhdCI6MTU3Nzk0MTMyNX0.ksgtQ8uDQ3322tOhQ4jIkzZjH6B0luhLA1NRKGCG00w",
      "user": Object {
        "__v": 0,
        "_id": "5e0bb5c61c90231c34596f83",
        "createdAt": "2019-12-31T20:55:34.424Z",
        "deleted": false,
        "email": "alexkim@dev.com",
        "hash": "8a68a2ca14e41925e745d3f5a14550a427aa05c4bc00248e63dc45d12e10e2e3f7e1e09822dd544c1f808dd71eb1aa6538b96408ff73d2ecfd9b54b204fad705",
        "name": "Alex the Great",
        "salt": "563ad7a2ec596b8bdca6cc5b3040e302",
        "updatedAt": "2019-12-31T21:00:15.298Z",
      },
    },
    "type": "users/register-success",
  },
]
`;

exports[`User Actions it should UPDATE a user 1`] = `
Array [
  Object {
    "type": "users/update-user-pending",
  },
  Object {
    "payload": Object {
      "user": Object {
        "__v": 0,
        "_id": "5e0bb5c61c90231c34596f83",
        "createdAt": "2019-12-31T20:55:34.424Z",
        "deleted": false,
        "email": "alexkim@dev.com",
        "hash": "8a68a2ca14e41925e745d3f5a14550a427aa05c4bc00248e63dc45d12e10e2e3f7e1e09822dd544c1f808dd71eb1aa6538b96408ff73d2ecfd9b54b204fad705",
        "name": "Alex Kim",
        "salt": "563ad7a2ec596b8bdca6cc5b3040e302",
        "updatedAt": "2019-12-31T20:55:34.424Z",
      },
    },
    "type": "users/update-user-success",
  },
]
`;
